graphics/src/AssimpLoader.cc

```diff
@@ -489,6 +530,9 @@ Mesh *AssimpLoader::Load(const std::string &_filename)
       // TODO parse different skeletons and merge them
       this->dataPtr->RecursiveSkeletonCreate(rootNode->mChildren[childIdx], rootSkelNode, rootTransform);
     }
+    rootSkelNode = rootSkelNode->Child(0); //We dont need scene node and adding will create inverse model
```

Comment:
Does not look safe to me? But if this is not the case, then whole logic is wrong.

Reply:
Agree, I think this is a good time to start investigating the TODO above about meshes with multiple skeletons.
A classic example we have been using internally is the one below, where the human and the cane have two separate skeletons that are merged in the "legacy" ColladaLoader.

[PatientWalkingCane.zip](https://github.com/gazebosim/gz-common/files/8963136/PatientWalkingCane.zip)

This model segfaults both with this PR and without and by looking at the stdout debug output from `assimp` it seems it might actually be an assimp parsing issue since it reports a lot of errors about bones not being found.
It seems then that the fix might once again be on the assimp side, then if this approach works for that edge case I think we can keep this approach until we find a case for which it doesn't work.

Reply:
Hi @luca-della-vedova,
Currently in my branch I cant see the reports related to bones not being found. I had seen it one time, but I think I lost it after some branch changes, can you share me the assimp, and gz-common branches you use? Prefabably after you merged this one.

Reply:
I can change the solution from removing the parent node to adding an identity transformation matrix if there is non.

Reply:
I can change the solution from removing the parent node to adding an identity transformation matrix if there is non exist?

Reply:
> When I try to load the mesh with the assimp PR you created, the whole Gazebo justs stucks and prints malloc(): invalid size (unsorted)), and if I dont use that PR, then we see this errors related to bones not being found. Furthermore, when I use the old colladaLoader, then again I see the Malloc error as well... So maybe it is an error not related to loaders? Should we find another dae file with multiple skeletons?

That's a bit strange, the model works (or at least used to work under Fortress) with the default collada loader, I wonder if the issue is somewhere else? Can you check that it works with a binary Fortress installation just to be safe?

>  I can change the solution from removing the parent node to adding an identity transformation matrix if there is non exist?

That makes me feel a bit better, but I thought the transformation is always populated and is an identity for `Scene`? Is that not the case?

Reply:
I see, then I guess setting the inverse bind transform to an identity sounds like a reasonable approach to me

Reply:
Friendly ping for this @onurtore, I think we can merge it after this change and start to look into opening a PR against `main`

Reply:
Hi @luca-della-vedova ,
Adding inverse bind transform to the scene brings back the animation segmentation fault. So now I am not sure whether should we add that or not.

Btw, mesh works with Fortress however, I am porting everthing to Fortress to find whether assimp has problems or not.

Reply:
Hi @luca-della-vedova, 
Good news, this mesh also loaded with assimp in Fortress. The only problem was related to texture as you can see below. 

![1](https://user-images.githubusercontent.com/20031290/176513660-107cc1ee-26b9-40d5-a61c-59878711376d.png)
 
I am in favour of merge, but the next think I will do is the divide the CreateActor() function into subparts. I tested the createActor function with the static analysis tools and dividing into subparts looks like good idea.

Cyclomatic Complexity: 36
#Lines: 302

Reply:
Assigning child and setting parent null creates segmentation error in Edifice demo.

Reply:
Yea I'm not sure how stable this solution is. The issue is not in adding a child and setting a parent to null but rather accessing the hardcoded element at index 0 in cases in which the root node does not have any children (which happens very often for simple meshes).
I added the "if" statement to only execute this block when there are animations in the scene which mitigates the issue, since scene with animations will most likely have a number of nodes, but it would be good to remove the hardcoded value or make it safer